src/js/skillExtractor.js

```
@@ -0,0 +1,95 @@
+ */
+function extractSkills(text) {
+  if (!text || typeof text !== 'string') return [];
+  
+  const normalizedText = text.toLowerCase();
+  const matchedSkills = new Set();
+  
+  // Simple word boundary checking for exact matches
+  // For multi-word skills like "machine learning", we use indexOf or regex
+  TECH_DICTIONARY.forEach(skill => {
+    // Escape special characters in skill (e.g. C++, Node.js)
+    const escapedSkill = skill.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
```

ESLint

[error] Unnecessary escape character: \/.

(no-useless-escape)